src/colvarbias_abmd.cpp

```diff
@@ -0,0 +1,136 @@
+// -*- c++ -*-
+
+// This file is part of the Collective Variables module (Colvars).
+// The original version of Colvars and its updates are located at:
+// https://github.com/Colvars/colvars
+// Please update all Colvars source files before making any changes.
+// If you wish to distribute your changes, please submit them to the
+// Colvars repository at GitHub.
+
+
+#include "colvarbias_abmd.h"
+#include "colvarproxy.h"
+#include <iomanip>
+
+
+colvarbias_abmd::colvarbias_abmd(char const *key)
+  : colvarbias(key),
+    colvarbias_ti(key)
+{
+}
+
+
+int colvarbias_abmd::init(std::string const &conf)
+{
+  cvm::main()->cite_feature("ABMD bias");
+
+  int err = colvarbias::init(conf);
+  err |= colvarbias_ti::init(conf);
+  if (err != COLVARS_OK) return err;
+
+  enable(f_cvb_apply_force);
+
+  if (num_variables() != 1) {
+    return cvm::error("ABMD requires exactly one collective variable.\n", COLVARS_INPUT_ERROR);
+  }
+
+  if ( ! (variables(0))->is_enabled(f_cv_scalar) ) {
+    return cvm::error("ABMD colvar must be scalar.\n", COLVARS_INPUT_ERROR);
+  }
+
+  get_keyval(conf, "forceConstant", k);
+  get_keyval(conf, "decreasing", decreasing, decreasing);
+  get_keyval(conf, "stoppingValue", stopping_val);
+
+  return COLVARS_OK;
+}
+
+
+int colvarbias_abmd::update()
+{
+  if (!cvm::main()->proxy->simulation_running()) {
+    return COLVARS_OK;
+  }
+
+  colvar const *cv = variables(0);
+  cvm::real const val = cv->value().real_value;
+
+  if (!ref_initialized) {
+    ref_val = val;
+    ref_initialized = true;
+  }
+
+  // Compute sign factor to unify increasing and decreasing cases below
+  // less conditionals, more arithmetic
+  cvm::real const sign = decreasing ? -1. : 1.;
+  cvm::real const diff = (val - ref_val) * sign;
+
+  if ( diff > 0. ) {
+    colvar_forces[0] = 0.;
+    bias_energy = 0.;
+    if ( (ref_val-stopping_val) * sign <= 0. ) ref_val = val;
+  } else {
+    colvar_forces[0] = - sign * k * diff;
+    bias_energy = 0.5 * k * diff * diff;;
```

Comment:
The convention here is that `k` is always in the units of (energy)/(colvar unit)^2, regardless of the value of `width`. This makes sense since this is one-dimensional by design, but I'm adding a quick note to clarify that this is different from the `harmonic` bias.